sidecar/interface.go

```diff
@@ -0,0 +1,230 @@
+	// bid order was canceled by the taker.
+	StateCanceled State = 6
+)
+
+// Offer is a struct holding the information that a sidecar channel provider is
+// committing to when offering to buy a channel for the recipient. The sidecar
+// channel flow is initiated by the provider creating a ticket and adding its
+// signed offer to it.
+type Offer struct {
+	// Capacity is the channel capacity of the sidecar channel in satoshis.
+	Capacity btcutil.Amount
+
+	// PushAmt is the amount in satoshis that will be pushed to the
+	// recipient of the channel to kick start them with inbound capacity.
+	// If this is non-zero then the provider must pay for the push amount as
+	// well as all other costs from their Pool account. Makers can opt out
+	// of supporting push amounts when submitting ask orders so this will
+	// reduce the matching chances somewhat.
+	PushAmt btcutil.Amount
+
+	// SignPubKey is the public key for corresponding to the private key
+	// that signed the SigOfferDigest below and, in a later state, the
+	// SigOrderDigest of the Order struct.
+	SignPubKey *btcec.PublicKey
+
+	// SigOfferDigest is a signature over the offer digest, signed with the
+	// private key that corresponds to the SignPubKey above.
+	SigOfferDigest *btcec.Signature
+}
+
+// Recipient is a struct holding the information about the recipient of the
+// sidecar channel in question. 
+type Recipient struct {
+	// NodePubKey is the recipient nodes' identity public key that is
+	// advertised in the bid order.
+	NodePubKey *btcec.PublicKey
+
+	// MultiSigPubKey is a public key to which the recipient node has the
+	// private key to. It is the key that will be used as one of the 2-of-2
+	// multisig keys of the channel funding transaction output and is
+	// advertised in the bid order.
+	MultiSigPubKey *btcec.PublicKey
+}
+
+// Order is a struct holding the information about the sidecar bid order after
+// it's been submitted by the provider.
+type Order struct {
+	// BidNonce is the order nonce of the bid order that was submitted for
+	// purchasing the sidecar channel.
+	BidNonce order.Nonce
+
+	// SigOrderDigest is a signature over the order digest, signed with the
+	// private key that corresponds to the SignPubKey in the Offer struct.
+	SigOrderDigest *btcec.Signature
+}
+
+// Execution is a struct holding information about the sidecar bid order during
+// its execution.
+type Execution struct {
+	// PendingChannelID is the pending channel ID of the currently
+	// registered funding shim that was added by the recipient node to
+	// receive the channel.
+	PendingChannelID [32]byte
+}
+
+// Ticket is a struct holding all the information for establishing/buying a
+// sidecar channel. It is meant to be used in a PSBT like manner where each
+// participant incrementally adds their information according to their role and
+// the current step.
+type Ticket struct {
+	// ID is a pseudorandom identifier of the ticket.
+	ID [8]byte
```

Comment:
Any reason 8 and not 32?

Reply:
Mostly to save some space. This doesn't need to be cryptographically strong IMO, just some unique identifier to tell them apart at the UI level.

Reply:
Hmm, so then in this case the issuer is the one responsible for uniqueness?

Reply:
I guess in the end, the nonce of the order is the thing that really binds it all.